pkg/crc/segment/segment_test.go

```diff
@@ -84,7 +86,8 @@ func TestClientUploadWithConsent(t *testing.T) {
 	c, err := newCustomClient(config, filepath.Join(dir, "telemetry"), server.URL)
 	require.NoError(t, err)
 
-	require.NoError(t, c.Upload(context.Background(), "start", time.Minute, errors.New("an error occurred")))
+	err = errors.New("an error occurred")
+	require.NoError(t, c.Upload(context.Background(), "start", time.Minute, fmt.Errorf("%w", err)))
```

Comment:
@cfergeau so I need to change the test to create the error change so unwrap can happen. Since `errors.New` doesn't implement `Unwrap` method from the interface side but `fmt.Errorf` have it.

Also if we are returning an `errors.New` any of upper caller function then on the segment the type would be `<nil>`.

Reply:
Checking if the error is `SerializableError` before unwrapping should be good enough for now? Or try to `Unwrap()`, and if the error is `nil`, return the initial error instead.